polyfill/lib/calendar.mjs

```diff
@@ -753,7 +753,6 @@ const nonIsoHelperBase = {
     // If the initial guess is not in the same month, then then bisect the
     // distance to the target, starting with 8 days per step.
     let increment = 8;
-    let maybeConstrained = false;
```

Comment:
Why remove the reverse mapping? Is it inaccurate sometimes?

Reply:
Yes, it was inaccurate in the case where the day needed to be constrained, but wasn't greater than `minimumMonthLength`. For example in the Hebrew calendar, 5781-M02-30 would erroneously cache the reverse mapping because that M02 has only 29 days in that year, but has 30 days in some other years. But that clamping already happened before the loop where `maybeConstrained` was checked, so the code assumed it was OK to cache the resulting ISO date back to 5781-M02-30.

It might be possible to rewrite this in such a way that it would be possible to catch this case and still cache the reverse mapping in other cases, but I don't have time to investigate that right now.